src/multidimensional_matrix_operators.jl

```diff
@@ -0,0 +1,198 @@
+
+"""
+    mass_matrix_boundary(D::MultidimensionalMatrixDerivativeOperator, dim)
+
+Construct the mass matrix at the boundary of a `MultidimensionalMatrixDerivativeOperator` `D` in direction `dim`.
+The boundary mass matrix is constructed to be mimetic, see
+
+- Jan Glaubitz, Simon-Christian Klein, Jan Nordström, Philipp Öffner (2023)
+  Multi-dimensional summation-by-parts operators for general function spaces: Theory and construction
+  Journal of Computational Physics 491, 112370, [DOI: 10.1016/j.jcp.2023.112370](https://doi.org/10.1016/j.jcp.2023.112370).
+"""
+mass_matrix_boundary(D::MultidimensionalMatrixDerivativeOperator, dim::Int) = Diagonal(weights_boundary_scaled(D, dim))
+
+function get_weight_boundary(D::MultidimensionalMatrixDerivativeOperator, i::Int)
+    @unpack weights_boundary, on_boundary = D
+    N, _ = size(D)
+    @boundscheck begin
+        @argcheck 1 <= i <= N
+    end
+    if !on_boundary[i]
+        return zero(eltype(D))
+    end
+    j = sum(view(on_boundary, 1:i))
+    @inbounds ω = weights_boundary[j]
+    return ω
+end
+
+function get_weight_boundary_scaled(D::MultidimensionalMatrixDerivativeOperator, dim::Int, i::Int)
+    @unpack normals, on_boundary = D
+    if !on_boundary[i]
+        return zero(eltype(D))
+    end
+    ω = get_weight_boundary(D, i)
+    j = sum(view(on_boundary, 1:i))
+    ω * normals[j][dim]
+end
+
+function Base.show(io::IO, D::MultidimensionalMatrixDerivativeOperator)
+    if get(io, :compact, false)
+        summary(io, D)
+    else
+        x = grid(D)
+        print(io, ndims(D), "-dimensional matrix-based first-derivative operator {T=", eltype(D), "}")
+        print(io, " on ", length(x), " nodes")
+    end
+end
+
+# TODO: mul! How? Depends on direction
+
+"""
+    tensor_product_operator_2D(D_x, D_y = D_x)
+
+Create a 2D [`MultidimensionalMatrixDerivativeOperator`](@ref) on a square based on a 1D derivative operators `D_x` and `D_y` using a tensor product structure.
+The operator `D_x` is used in the x-direction and `D_y` in the y-direction.
+
+For the construction, see also:
+- Sigrun Ortleb (2021)
+  Numerical Methods for Fluid Flow:
+    High Order SBP Schemes, IMEX Advection-Diffusion Splitting and Positivity Preservation for Production-Destruction-PDEs
+  Habilitation thesis, University of Kassel, [DOI: 10.17170/kobra-202301037274](https://doi.org/10.17170/kobra-202301037274),
+  Chapter 1.2.4.
+- Magnus Svärd, Jan Nordström (2014)
+  Review of summation-by-parts schemes for initial-boundary-value problems
+  Journal of Computational Physics 268, pp. 17-38, [DOI: 10.1016/j.jcp.2014.02.031](https://doi.org/10.1016/j.jcp.2014.02.031).
+"""
+function tensor_product_operator_2D(D_x, D_y = D_x)
+    T = promote_type(eltype(D_x), eltype(D_y))
+    nodes_1D_x = grid(D_x)
+    nodes_1D_y = grid(D_y)
+    N_x = length(nodes_1D_x)
+    N_y = length(nodes_1D_y)
+    nodes = SVector.(vec(nodes_1D_x' .* ones(N_y)), vec(ones(N_x)' .* nodes_1D_y))
```

Comment:
`ones` creates `Float64` by default. Can we please preserve the type of the coefficients - even if the code becomes a bit longer and may not fit on a single line anymore?